cmd/dist/common.go

```diff
@@ -12,10 +17,31 @@ import (
 	"github.com/containerd/containerd/remotes"
 	"github.com/containerd/containerd/remotes/docker"
 	imagesservice "github.com/containerd/containerd/services/images"
+	"github.com/crosbymichael/console"
+	"github.com/pkg/errors"
 	"github.com/urfave/cli"
 	"google.golang.org/grpc"
 )
 
+var registryFlags = []cli.Flag{
+	cli.BoolFlag{
+		Name:  "insecure,k",
```

Comment:
`insecure` is a bit of a misnomer that can drive one insane by tying too much behavior to it. Let's make these properly granular.

The interesting example would be `https` vs `http` and `insecure`. What happens? Even though I configure `https`, does that mean we change the protocol or just don't verify? With `skip-verify`, we make it clear: protocol selects the protocol and this controls the verification. Arguably, there should also be a `tls` flag but I don't think that is necessary here.